pytorch_lightning/loggers/comet.py

```diff
@@ -229,7 +229,8 @@ def log_metrics(
             if is_tensor(val):
                 metrics[key] = val.cpu().detach()
 
-        self.experiment.log_metrics(metrics, step=step)
+        epoch = metrics.pop('epoch', None)
```

Comment:
could you explain why pop and not get?

Reply:
I think that sending epoch value as a metric is kinda strange, because it is not. Moreover if we send both epoch as an argument and epoch as metric value they will be duplicated on the comet dashboard.
Although changing this to get will do the trick

Reply:
Lightning plots epoch against global step in every logger, it's a good thing to have, because you will know how many steps one epoch corresponds to, especially in cases where epochs have different size.
Then, in your logging UI, you can choose epoch as the abscissa in your plots, since it is monotonic.

Reply:
if you don't want it in comet, make a copy of the metrics dict, and pop it. maybe that works.

Reply:
if comet has a special keyword for epoch already, maybe the solution is to copy the dict, pop epoch and pass the copy to the logger.

Reply:
Ok, got your point about other loggers. As you've said comet can handle epoch separately. I'll do the copying way